content/2019-02-20-newsletter-15.md

```diff
@@ -0,0 +1,191 @@
+
+- Math support in core
+
+That is `0f32.sin()` should "Just Work" in `no_std` crates. There are a few questions about how to best implement this without degrading performance of applications that link to `std` (we want those to use the arch optimized routines in e.g. glibc's libm instead of the generic Rust implementation) and we need a champion to do the research.
+
+- Cargo build hooks
+
+This refers to the ability to run custom code after `cargo build`. We covered use cases, their requirements and discussed the trade-offs of a more general Cargo tasks (e.g. `cargo $task`) mechanism vs a more targeted post-build script (e.g. `post-build.rs`) mechanism. Expect more news (a (pre-)RFC) from the WASM WG on this front.
+
+- Slimmer `core::fmt`.
+
+The current implementation of `core::fmt` uses trait objects and function pointers to make all uses of `core::fmt` fast to compile but this makes `core::fmt` impossible to inline which makes `no_std` programs that use formatting large in (binary) size. It might be possible to leverage std-aware Cargo to fix this: we could add a Cargo feature to `core` to replace the current implementation with one that's fully inlineable but as featureful as the current one -- this should produce smaller binaries.
+
+For a bit more of detail you can check the [logs] and [minutes] from two meetings ago; they include our notes from the All Hands
+
+[logs]: https://github.com/rust-embedded/wg/blob/master/minutes/2019-02-12.irc.log
+[minutes]: https://github.com/rust-embedded/wg/blob/master/minutes/2019-02-12.md#notes-from-rust-all-hands
+
+## Highlights
+
+- The Embedded WG has started receiving [submissions] for [the embedded showcase][showcase]! We are looking for cool embedded Rust projects with visuals (pictures or videos) that can serve as examples of what can be done in Rust.
+
+[submissions]: https://github.com/rust-embedded/showcase#submit-your-project
+[showcase]: https://github.com/rust-embedded/showcase
+
+- The [stm32-rs] organization has emerged to collate work on STM32 crates.
+
+[stm32-rs]: https://github.com/stm32-rs
+
+- Similarly, the [lpc-rs] organization has emerged to work on LPC microcontrollers by NXP.
+
+[lpc-rs]: https://github.com/lpc-rs
+
+- [@adamgreig] has released an experimental PAC (Peripheral Access Crate) alternative for STM32 microcontrollers: [stm32ral] that compiles in about 3 seconds. (PACs are known for compile times in the order of hundreds of seconds.)
+
+[@adamgreig]: https://github.com/adamgreig
+[stm32ral]: https://github.com/adamgreig/stm32ral
+
+- As of `nightly-2019-02-05` the [embedded Rust Book][book] is included as part of the docs shipped with the Rust toolchain!
```

Comment:
Consider linking to the nightly docs: https://doc.rust-lang.org/nightly/#embedded-systems